vault/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:22.04
+
+WORKDIR /vault
+
+RUN apt update -y && apt install wget gpg -y && \
+    wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg && \
+    gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint && \
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com focal main" | tee /etc/apt/sources.list.d/hashicorp.list && \
```

Comment:
Could be great to add a comment with some explanations of what this does (or a link to the documentation ?)